Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -706,4 +708,60 @@ GC::Ref<HTML::DOMStringList> create_a_sorted_name_list(JS::Realm& realm, Vector<
     return HTML::DOMStringList::create(realm, names);
 }
 
+// https://w3c.github.io/IndexedDB/#commit-a-transaction
+void commit_a_transaction(JS::Realm& realm, GC::Ref<IDBTransaction> transaction)
+{
+    // 1. Set transaction’s state to committing.
+    transaction->set_state(IDBTransaction::TransactionState::Committing);
+
+    dbgln_if(IDB_DEBUG, "commit_a_transaction: transaction {} is committing", transaction->uuid());
+
+    // 2. Run the following steps in parallel:
+    Platform::EventLoopPlugin::the().deferred_invoke(GC::create_function(realm.heap(), [&realm, transaction]() {
+        HTML::TemporaryExecutionContext context(realm, HTML::TemporaryExecutionContext::CallbacksEnabled::Yes);
+
+        // 1. Wait until every item in transaction’s request list is processed.
+        HTML::main_thread_event_loop().spin_until(GC::create_function(realm.vm().heap(), [transaction]() {
```

Comment:
no action, but it sure feels weird to spin the main event loop from "in parallel". I feel like this is a pattern we've done elsewhere, but it sure .. feels weird :man_shrugging:

Reply:
We could possibly spin the DatabaseAccess task queue instead..? But last i heard they are all the same queue, so maybe not? 🤷